ome_zarr/data.py

```diff
@@ -0,0 +1,166 @@
+    label_image = label(cleared)
+
+    pyramid = list(reversed([zoom(image, 2 ** i, order=3) for i in range(4)]))
+    labels = list(reversed([zoom(label_image, 2 ** i, order=0) for i in range(4)]))
+
+    pyramid = [rgb_to_5d(layer) for layer in pyramid]
+    labels = [rgb_to_5d(layer) for layer in labels]
+    return pyramid, labels
+
+
+def astronaut() -> Tuple[List, List]:
+    """Sample data from skimage."""
+    scaler = Scaler()
+
+    pixels = rgb_to_5d(np.tile(data.astronaut(), (2, 2, 1)))
+    pyramid = scaler.nearest(pixels)
+
+    shape = list(pyramid[0].shape)
+    shape[1] = 1
+    label = np.zeros(shape)
+    make_circle(100, 100, 1, label[0, 0, 0, 200:300, 200:300])
+    make_circle(150, 150, 2, label[0, 0, 0, 250:400, 250:400])
+    labels = scaler.nearest(label)
+
+    return pyramid, labels
+
+
+def make_circle(h: int, w: int, value: int, target: np.ndarray) -> None:
+    """Apply a 2D circular mask to the given array.
+
+    >>> import numpy as np
+    >>> example = np.zeros((8, 8))
+    >>> make_circle(8, 8, 1, example)
+    >>> print(example)
+    [[0. 0. 0. 0. 0. 0. 0. 0.]
+     [0. 0. 1. 1. 1. 1. 1. 0.]
+     [0. 1. 1. 1. 1. 1. 1. 1.]
+     [0. 1. 1. 1. 1. 1. 1. 1.]
+     [0. 1. 1. 1. 1. 1. 1. 1.]
+     [0. 1. 1. 1. 1. 1. 1. 1.]
+     [0. 1. 1. 1. 1. 1. 1. 1.]
+     [0. 0. 1. 1. 1. 1. 1. 0.]]
+    """
+    x = np.arange(0, w)
+    y = np.arange(0, h)
+
+    cx = w // 2
+    cy = h // 2
+    r = min(w, h) // 2
+
+    mask = (x[np.newaxis, :] - cx) ** 2 + (y[:, np.newaxis] - cy) ** 2 < r ** 2
+    target[mask] = value
+
+
+def rgb_to_5d(pixels: np.ndarray) -> List:
+    """Convert an RGB image into 5D image (t, c, z, y, x)."""
+    if len(pixels.shape) == 2:
+        stack = np.array([pixels])
+        channels = np.array([stack])
+    elif len(pixels.shape) == 3:
+        size_c = pixels.shape[2]
+        channels = [np.array([pixels[:, :, c]]) for c in range(size_c)]
+    else:
+        assert False, f"expecting 2 or 3d: ({pixels.shape})"
+    video = np.array([channels])
+    return video
+
+
+def write_multiscale(pyramid: List, group: zarr.Group) -> None:
+    """Write a pyramid with multiscale metadata to disk."""
+    paths = []
+    for path, dataset in enumerate(pyramid):
+        group.create_dataset(str(path), data=pyramid[path])
+        paths.append({"path": str(path)})
+
+    multiscales = [{"version": "0.1", "datasets": paths}]
+    group.attrs["multiscales"] = multiscales
+
+
+def create_zarr(
+    zarr_directory: str,
+    method: Callable[..., Tuple[List, List]] = coins,
+    label_name: str = "coins",
+) -> None:
+    """Generate a synthetic image pyramid with labels."""
+    pyramid, labels = method()
+
+    store = zarr.DirectoryStore(zarr_directory)
+    grp = zarr.group(store)
+    write_multiscale(pyramid, grp)
+
+    if pyramid[0].shape[CHANNEL_DIMENSION] == 1:
+        image_data = {
+            "channels": [{"window": {"start": 0, "end": 1}}],
+            "rdefs": {"model": "grayscale"},
```

Comment:
In the OMERO metadata JSON I think we previously had "greyscale" matching `RenderingModel`. I don't know if this matters or what it should be.

Reply:
Happy to defer to @will-moore, but I agree at least the three zarr related repos should use uniform spelling.

Reply:
Yup. OMERO has `greyscale` everywhere. napari `grayscale`. I'll update to use the former and if we need to, we can transform in `ome_zarr.napari`.

NB: ome-model primarily uses "e", "a" appears only twice in docs. Bio-Formats leans more towards "a" including the CLI argument.